ironfish-cli/STYLE_GUIDE.md

```diff
@@ -0,0 +1,68 @@
+
+Because topics are generally nouns, the root command of a topic usually lists those nouns. So in the case of `ironfish migrations`, it will list all the migrations. Never create a *:list command such as `ironfish migrations:list`.
+
+## Command Description
+Topic and command descriptions should be provided for all topics and commands. They should fit on 80 character width screens, begin with a lowercase character, and should not end in a period.
+
+## Input
+Input to commands is typically provided by flags and args. Stdin can also be used in cases where it is useful to stream files or information in (such as with heroku run).
+
+### Flags
+
+Flags are preferred to args when there are many inputs, particularly inputs of the same type. They involve a bit more typing, but make the use of the CLI clearer. For example, `ironfish wallet:send` used to accept an argument for the account to use, as well as --to flag to specify the account to send to.
+
+So using heroku fork used to work like this:
+
+```bash
+$ ironfish wallet:send source_account --to dest_account
+```
+
+This is confusing to the user since it isn’t clear which account they are sending from and which one they are sending to. By switching to required flags, we instead expect input in this form:
+
+```bash
+ironfish wallet:send --account source_account --to dest_account
+```
+
+This also allows the user to specify the flags in any order, and gives them the confidence that they are running the command correctly. It also allows us to show better error messages.
+
+Ensure that descriptions are provided for all flags, that the descriptions are in lowercase, that they are concise (so as to fit on narrow screens), and that they do not end in a period to match other flag descriptions.
+
+
+### Arguments
+Arguments are the basic way to provide input for a command. While flags are generally preferred, they are sometimes unnecessary in cases where there is only 1 argument, or the arguments are obvious and in an obvious order.
+
+In the case of `ironfish chain:power`, we can specify the account we want to give access to with an argument.
+
+```bash
+ironfish chain:power 5432 --blocks 20
+```
+
+If this was done with only arguments, it wouldn’t be clear if the sequence should go before or after the number of blocks to use to sample the network mining power. Using a required flag instead allows the user to specify it either way.
+
+### Prompting
+Prompting for missing input provides a nice way to show complicated options in the CLI. For example, `ironfish wallet:use` shows the following if no account is specified as an arg.
+
+```
+$ ironfish wallet:use
+
+? Which wallet would you like to use? (Use arrow keys)
```

Comment:
`wallet:use` doesnt have this prompt right now, right?

Reply:
It's more of a guideline for how it should be. I want to be able to point people to this guide and say if you wanna make improve UX you can go make commands adhere to this guide